sql/core/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -690,6 +718,7 @@ class KeyValueGroupedDataset[K, V] private[sql](
    * @param initialState      User provided initial state that will be used to initiate state for
    *                          the query in the first batch.
    *
+   * See [[Encoder]] for more details on what types are encodable to Spark SQL.
    */
   private[sql] def transformWithState[U: Encoder, S: Encoder](
```

Comment:
Is this an API? we should remove `private[sql]`

Reply:
Same as above. We have a bunch of sub-features that we are working on that might need small changes to the APIs here. So we have kept this as `package private` for now and will make it public once all the changes are finalized.
cc - @HeartSaVioR

Reply:
We want to open everything at once when we are ready to release the feature, hence we guard the API via `private[sql]`.